trade-finance-manager-ui/component-tests/utilisation-reports/_macros/payment-group-radio-input.component-test.ts

```diff
@@ -0,0 +1,60 @@
+import { AvailablePaymentGroupsViewModel } from '../../../server/types/view-models/add-to-an-existing-payment-view-model';
+import { componentRenderer } from '../../componentRenderer';
+import { anAvailablePaymentGroupsViewModel } from '../../../test-helpers/test-data/available-payment-groups-view-model';
+
+const component = '../templates/utilisation-reports/_macros/payment-group-radio-input.njk';
+const render = componentRenderer(component);
+
+describe(component, () => {
+  const getWrapper = (viewModel: { legendText: string; paymentGroups: AvailablePaymentGroupsViewModel }) => render(viewModel);
+
+  it('should render the legend text', () => {
+    const viewModel = {
+      legendText: 'Select a payment group',
+      paymentGroups: [],
+    };
+    const wrapper = getWrapper(viewModel);
+
+    wrapper.expectText('h2[data-cy="payment-group-radio-input-heading"]').toContain('Select a payment group');
+  });
+
+  it('should render radio buttons for each payment group with group ID', () => {
+    const viewModel = {
+      legendText: 'Select a payment group',
+      paymentGroups: anAvailablePaymentGroupsViewModel(),
+    };
+    const wrapper = getWrapper(viewModel);
+
+    const radioInputSelector = 'div.govuk-radios__item > input[type="radio"].govuk-radios__input';
+    wrapper.expectElement(radioInputSelector).toHaveCount(viewModel.paymentGroups.length);
+
+    wrapper.expectElement(`${radioInputSelector}:eq(0)`).toHaveAttribute('id', 'paymentIds-1,2');
+    wrapper.expectElement(`${radioInputSelector}:eq(0)`).toHaveAttribute('value', 'paymentIds-1,2');
+
+    wrapper.expectElement(`${radioInputSelector}:eq(1)`).toHaveAttribute('id', 'paymentIds-3');
+    wrapper.expectElement(`${radioInputSelector}:eq(1)`).toHaveAttribute('value', 'paymentIds-3');
+  });
+
+  it('should render multiple payment details within a group', () => {
```

Comment:
The test name doesn't match the assertion, in the sense that we are also testing the single payment case